homeassistant/components/vesync/coordinator.py

```diff
@@ -0,0 +1,37 @@
+"""Class to manage VeSync data updates."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pyvesync import VeSync
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VeSyncDataCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class representing data coordinator for VeSync devices."""
+
+    def __init__(self, hass: HomeAssistant, manager: VeSync) -> None:
+        """Initialize."""
+        self._manager = manager
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="VeSyncDataCoordinator",
+            update_interval=timedelta(seconds=UPDATE_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Fetch data from API endpoint."""
+
+        # Using `update_all_devices` instead of `update` to avoid fetching device list every time.
+        return await self.hass.async_add_executor_job(self._manager.update_all_devices)
```

Comment:
Do you know if this call also update energy for plugs?  Since I think that is a separate call? Should entity update also be removed from switch.py?

Reply:
I assumed that `update` took care of that but that doesn't seem to be the case for outlets. Do you know why it might been designed as a separate thing? I can invoke a second job for `VeSync.update_energy` to handle that though. Thoughts?

Reply:
Yes I think just calling update energy should do it!

Reply:
Okay fixed that too. I ended up writing more unit tests. The tests can be improved, there is some duplicate content but I think it would be okay to do that in separate PR. My tests kept failing till I discovered that energy updates are only permitted every 6 hours.